### ISSUE TITLE
Microsoft.VisualBasic.dll has no FileVersion in non-Windows transport packages

### ISSUE THREAD

**Issue by dagood** [dotnet/corefx#36630]
In Core-Setup, I'm adding a check to make sure all DLL files in the Platform Manifest have `FileVersion`s. (https://github.com/dotnet/core-setup/pull/5682) I hit a problem in non-Windows PR validation with `Microsoft.VisualBasic.dll` not having versions in non-Windows transport packages:

| File in package cache | FileVersion |
| --- | --- |
| microsoft.private.corefx.netcoreapp\4.6.0-preview4.19204.3\ref\netcoreapp3.0\Microsoft.VisualBasic.dll | 14.7.3060.0 |
| runtime.linux-arm.microsoft.private.corefx.netcoreapp\4.6.0-preview4.19204.3\runtimes\linux-arm\lib\netcoreapp3.0\Microsoft.VisualBasic.dll | **0.0.0.0** |
| runtime.linux-arm64.microsoft.private.corefx.netcoreapp\4.6.0-preview4.19204.3\runtimes\linux-arm64\lib\netcoreapp3.0\Microsoft.VisualBasic.dll | **0.0.0.0** |
| runtime.linux-musl-x64.microsoft.private.corefx.netcoreapp\4.6.0-preview4.19204.3\runtimes\linux-musl-x64\lib\netcoreapp3.0\Microsoft.VisualBasic.dll | **0.0.0.0** |
| runtime.linux-x64.microsoft.private.corefx.netcoreapp\4.6.0-preview4.19204.3\runtimes\linux-x64\lib\netcoreapp3.0\Microsoft.VisualBasic.dll | **0.0.0.0** |
| runtime.osx-x64.microsoft.private.corefx.netcoreapp\4.6.0-preview4.19204.3\runtimes\osx-x64\lib\netcoreapp3.0\Microsoft.VisualBasic.dll | **0.0.0.0** |
| runtime.rhel.6-x64.microsoft.private.corefx.netcoreapp\4.6.0-preview4.19204.3\runtimes\rhel.6-x64\lib\netcoreapp3.0\Microsoft.VisualBasic.dll | **0.0.0.0** |
| runtime.win-arm.microsoft.private.corefx.netcoreapp\4.6.0-preview4.19204.3\runtimes\win-arm\lib\netcoreapp3.0\Microsoft.VisualBasic.dll | 14.7.3060.0 |
| runtime.win-arm64.microsoft.private.corefx.netcoreapp\4.6.0-preview4.19204.3\runtimes\win-arm64\lib\netcoreapp3.0\Microsoft.VisualBasic.dll | 14.7.3060.0 |
| runtime.win-x64.microsoft.private.corefx.netcoreapp\4.6.0-preview4.19204.3\runtimes\win-x64\lib\netcoreapp3.0\Microsoft.VisualBasic.dll | 14.7.3060.0 |
| runtime.win-x86.microsoft.private.corefx.netcoreapp\4.6.0-preview4.19204.3\runtimes\win-x86\lib\netcoreapp3.0\Microsoft.VisualBasic.dll | 14.7.3060.0 |

Other DLLs have correct FileVersions in the non-Windows runtime transport packages, including `Microsoft.VisualBasic.Core.dll`.

/cc @ericstj

**Comment by ericstj** [comment]
It actually has a file-version but that file version is 0.0.0.0.
![image](https://user-images.githubusercontent.com/8918108/55641974-10c29d00-5784-11e9-8413-a2359bf9d91f.png)

I wonder if this is due to a bad reference assembly used to build the shim.  I'll investigate.

**Comment by ericstj** [comment]
Funny bug.  So the Microsoft.VisualBasic.dll reference assembly in the .NETFramework targeting pack has a  file version in the PE version resources but is missing the AssemblyFileVersion attribute in its metadata.  Even the implementation DLL in desktop is missing the AssemblyFileVersion attribute.

When GenFacades (rewriter) writes the facade for this shim it attempts to read the fileversion from the contract assembly then write it to the facade assembly it's generating.
https://github.com/dotnet/arcade/blob/5b8a27d82fc3ed08196ff40d42401931e345d406/src/Microsoft.DotNet.GenFacades/GenFacades.cs#L668-L683

On Unix our implementation of FileVersionInfo attempts to read from metadata and not PE version resources.  
https://github.com/dotnet/corefx/blob/c575bed77090d8e37b466197151acc3c36a959b6/src/System.Diagnostics.FileVersionInfo/src/System/Diagnostics/FileVersionInfo.Unix.cs#L20-L28

As such gen-facades gets out 0.0.0.0 (since that's what the FileVersionInfo returns when no version is found) and then writes that back.

A few ways to fix this: 
1. Add PE reader to either GenFacades or FileVersionInfo..
2. Switch these facades to use the source-generator version of GenFacades.  Potentially just create a single manual facade project for Microsoft.VisualBasic that does this.

/cc @Anipik

**Comment by danmosemsft** [comment]
@ericstj https://github.com/dotnet/corefx/issues/31900

**Comment by Anipik** [comment]
I just verified and the version number for all these transport packages(for windows and non-windows pacakages) on preview 7 is non zero
This must have been fixed when we  started use sourceGenerator for generating all the facades.
![Capture](https://user-images.githubusercontent.com/10516582/59310278-c1548d80-8c5a-11e9-9e8b-1121ec58fc02.JPG)

The attached image is for rhel. I verified for linux as well. The version value is  exactly same on windows and linux
